staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -388,12 +388,13 @@ func interfaceFromUnstructured(sv, dv reflect.Value) error {
 	return nil
 }
 
-func (c *converterImpl) ToUnstructured(obj interface{}, u *map[string]interface{}) error {
+func (c *converterImpl) ToUnstructured(obj interface{}) (map[string]interface{}, error) {
```

Comment:
A little confuse here, so actually this function doesn't covert an interface obj to map[string]interface{} ? if so why should we return a empty allocated map, I think that also can cause confusing ?